src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1513,15 +1513,12 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
         deltaVelocitySquared = 0.0_RKIND
         do i = 1, nEdgesOnCell(iCell)
           iEdge = edgesOnCell(i, iCell)
-          factor = 0.5_RKIND * dcEdge(iEdge) * dvEdge(iEdge) * invAreaCell
           delU2 =  (surfaceStress(iEdge))**2
-          deltaVelocitySquared = deltaVelocitySquared + factor * delU2
+          deltaVelocitySquared = deltaVelocitySquared + edgeAreaFractionOfCell(i,iCell) * delU2
         enddo
-        surfaceStressMagnitude(iCell) = sqrt(deltaVelocitySquared)
-
-       ! compute surface friction velocity
-         surfaceFrictionVelocity(iCell) = sqrt(surfaceStressMagnitude(iCell) / rho_sw)
-
+       ! NOTE that the factor of 2 is from averaging dot products to cell centers on a C-grid
+        surfaceStressMagnitude(iCell) = sqrt(2.0_RKIND * deltaVelocitySquared )
+        surfaceFrictionVelocity(iCell) = surfaceStressMagnitude(iCell) / rho_sw
```

Comment:
I don't think either this version or as originally listed in the code is right.  Let me look at this more deeply.